### PR TITLE
feat: added random password feature

### PR DESCRIPTION
**Pull request by gabisavu-ionos**
## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

**Comment by cristiGuranIonos** [comment]
Good! You still need to do the same for `image_password` in server resources.

File: ionoscloud/resource_user_test.go [hunk 1]
```diff
@@ -184,44 +190,72 @@ func testAccCheckUserExists(n string, user *ionoscloud.User) resource.TestCheckF
...
+}
+
 resource ` + UserResource + ` ` + UserTestResource + ` {
  first_name = "` + UpdatedResources + `"
  last_name = "` + UpdatedResources + `"
  email = "` + utils.GenerateEmail() + `"
- password = "abc123-321CBAupdated"
+ password 		= ` + RandomPassword + `.user_password_updated.result
+#password = ` + RandomPassword + `.user_password.result Updated
+ #password = join("Updated", ` + RandomPassword + `.user_password_updated.result)
```
[the remark this answers is not in the document]
Comment (gabisavu-ionos) [reply ionoscloud/resource_user_test.go:240]: yes, should be removed

File: ionoscloud/resource_user_test.go [hunk 2]
```diff
@@ -86,7 +89,10 @@ func TestAccUserBasic(t *testing.T) {
 					resource.TestCheckResourceAttrSet(UserResource+"."+UserTestResource, "email"),
 					resource.TestCheckResourceAttr(UserResource+"."+UserTestResource, "administrator", "false"),
 					resource.TestCheckResourceAttr(UserResource+"."+UserTestResource, "force_sec_auth", "false"),
-					resource.TestCheckResourceAttr(UserResource+"."+UserTestResource, "active", "false")),
+					resource.TestCheckResourceAttr(UserResource+"."+UserTestResource, "active", "false"),
+					resource.TestCheckResourceAttrPair(UserResource+"."+UserTestResource, "password", RandomPassword+".user_password_updated", "result"),
+					//resource.TestCheckResourceAttr(UserResource+"."+UserTestResource, "password", RandomPassword+".user_password.result"+"Updated"),
```
[the remark this answers is not in the document]
Comment (gabisavu-ionos) [reply ionoscloud/resource_user_test.go:94]: done

**Review by adeatcu-ionos — CHANGES_REQUESTED** [review]

File: docs/index.md [hunk 3]
```diff
@@ -254,6 +254,11 @@ resource "ionoscloud_server" "example" {
     }
 }
 
+resource "random_password" "server_image_password" {
```
[the remark this answers is not in the document]
Comment (gabisavu-ionos) [reply docs/index.md:257]: yes, in thee test, the password format is different, my bad, I forgot to update the docs too, I'll search every place where image password is not updated

File: ionoscloud/data_source_servers_test.go [hunk 4]
```diff
@@ -319,6 +329,15 @@ resource ` + ServerResource + ` ` + serverTestResource2 + ` {
   }
 }
 
+resource ` + RandomPassword + ` "server_image_password" {
```
[the remark this answers is not in the document]
Comment (gabisavu-ionos) [reply ionoscloud/data_source_servers_test.go:332]: server2_image_password is used only in this file, but for server_image_password, makes sense

File: docs/resources/volume.md [hunk 5]
```diff
@@ -88,9 +88,20 @@ resource "ionoscloud_volume" "example" {
...
+  image_password          = random_password.volume_image_password.result
   user_data               = "foo"
 }
+
+resource "random_password" "server_image_password" {
+  length           = 16
+  special          = false
+}
+
+resource "random_password" "volume_image_password" {
```
Comment (adeatcu-ionos) [inline docs/resources/volume.md:100]: @gabisavu-ionos the `volume_image_password` isn't the same as the `server_image_password`? I'm looking into the `resource_server.go` and it seems like the `image_password` is the same for both server and volume. Can you please double check this, please? I think that for the `volume_image_password` we should also have `length = 16` and `special = false`.

**Review by adeatcu-ionos — CHANGES_REQUESTED** [review]

File: docs/resources/volume.md (hunk 5 above)
Comment (gabisavu-ionos) [reply docs/resources/volume.md:100]: indeed is, I changed it

**Review by adeatcu-ionos — APPROVED** [review]

**Review by cristiGuranIonos — APPROVED** [review]

**Pull request merged**